### PR TITLE
SVG Preprocessor

### LINKED ISSUE #168: SVG normalizer
https://github.com/Jack000/Deepnest/issues/191 will fix this with a rust plugin, that normalizes the SVG so it has everything our current state can work with.

PR will follow maybe this or next week.

### PR DESCRIPTION
**Pull request by Dexus**
Convert all to SVG Path elements.

no missing circles, not missing other parts anymore.

fixes #168

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
   "dependencies": {
     "@electron/remote": "^2.1.2",
     "graceful-fs": "^4.2.11",
+    "@deepnest/svg-preprocessor": "^0.1.1",
```
Comment (abebeos) [inline package.json:40]: Where is the source-code of this?

**Comment by abebeos** [comment]
> ?

https://github.com/deepnest-io/nodejs-svg-preprocessor

**Comment by Dexus** [comment]
Right the code can be found currently in the private repo. It will make the SVG elements as a flat svg with only path elements. This helps to make sure the circle keep the position and also non supported elements. 

I had no difficulties with it but maybe - there are other problems that I not yet have seen. Maybe we make it optional? Because of the behavior?

For my svgs it had done a great job.

**Comment by ShaMan123** [comment]
Does it handle text? I wrote a text plugin in my branch using opentype.js

**Comment by abebeos** [comment]
Just be aware: at least some people use this software. Fixing one bug / optimizing performance, and introducing hidden bugs:

=> bad.

we don't have an automated test-system, so manual testing by different users must do it.

**Comment by Dexus** [comment]
Thats true, new bugs would be bad, always. 

But with the current code style we will, if we dont make changes, never be able to get modern code. That is also testable. I think once we have a more modern base, also moved from pure js to typescript we will have a good part that will also be testable. It should be our focus to have the code more clean and documented. So its eill be easier for everyone to code and test it.

**Comment by abebeos** [comment]
* https://github.com/deepnest-io/Deepnest/issues/174

**Comment by abebeos** [comment]
@Dexus, all this looks good. We can include this after stabilizing an electron >30 release.

**Comment by abebeos** [comment]
please open a new PR for this

**Comment by Dexus** [comment]
Why open a new one? I can easy rebase it and create a force Pull to the branch. No need to create a new PR.

**Comment by abebeos** [comment]
> Why open a new one?

this is how I deal with sensory-overload (e.g. when too many things go wrong).

> I can easy rebase it and create a force Pull to the branch. 

why didn't you showcase it immediately?

**Pull request closed without merging**